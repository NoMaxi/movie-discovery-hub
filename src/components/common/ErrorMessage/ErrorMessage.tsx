interface ErrorMessageProps {
    message: string;
    onDismiss: () => void;
    className?: string;
}

export const ErrorMessage = ({ message, onDismiss, className = "" }: ErrorMessageProps) => {
    if (!message) {
        return null;
    }

    return (
        <div data-testid="error-container" className={`flex items-center text-center ${className}`}>
            <span data-testid="error-message" className="text-red-500 mr-3">
                {message}
            </span>
            <button
                data-testid="error-dismiss-button"
                onClick={onDismiss}
                className="
                    text-3xl text-red-300 hover:text-[var(--color-primary)]
                    transition-colors duration-300 cursor-pointer
                "
                aria-label="Dismiss error"
            >
                &times;
            </button>
        </div>
    );
};
