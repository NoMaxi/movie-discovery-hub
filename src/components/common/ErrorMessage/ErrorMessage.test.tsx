import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent, { UserEvent } from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { ErrorMessage } from "./ErrorMessage";

describe("ErrorMessage", () => {
    const mockOnDismiss = jest.fn();
    const testMessage = "Test error message";
    let user: UserEvent;

    beforeEach(() => {
        mockOnDismiss.mockClear();
        user = userEvent.setup();
    });

    const renderComponent = (props: Partial<React.ComponentProps<typeof ErrorMessage>> = {}) => {
        const defaultProps: React.ComponentProps<typeof ErrorMessage> = {
            message: testMessage,
            onDismiss: mockOnDismiss,
        };
        const mergedProps = { ...defaultProps, ...props };
        return render(<ErrorMessage {...mergedProps} />);
    };

    it("Should render the error message correctly", () => {
        const { asFragment } = renderComponent();

        expect(screen.getByTestId("error-message")).toHaveTextContent(testMessage);
        expect(screen.getByLabelText("Dismiss error")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    it("Should call onDismiss when dismiss button is clicked", async () => {
        renderComponent();
        const dismissButton = screen.getByTestId("error-dismiss-button");

        await user.click(dismissButton);

        expect(mockOnDismiss).toHaveBeenCalledTimes(1);
    });

    it("Should apply className prop correctly", () => {
        const testClass = "test-class";
        renderComponent({ className: testClass });
        const container = screen.getByTestId("error-container");

        expect(container).toHaveClass(testClass);
        expect(container).toHaveClass("flex");
        expect(container).toHaveClass("items-center");
        expect(container).toHaveClass("text-center");
    });

    it("Should return null when no message is provided", () => {
        const { container } = renderComponent({ message: "" });

        expect(container).toBeEmptyDOMElement();
    });

    it("Should render the message in red text", () => {
        renderComponent();
        const messageElement = screen.getByTestId("error-message");

        expect(messageElement).toHaveClass("text-red-500");
    });

    it("Should have the correct button styling", () => {
        renderComponent();
        const button = screen.getByTestId("error-dismiss-button");

        expect(button).toHaveClass("text-3xl");
        expect(button).toHaveClass("text-red-300");
        expect(button).toHaveClass("transition-colors");
        expect(button).toHaveClass("duration-300");
        expect(button).toHaveClass("cursor-pointer");
    });
});
