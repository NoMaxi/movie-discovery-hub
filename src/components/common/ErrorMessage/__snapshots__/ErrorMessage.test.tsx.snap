// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Should render the error message correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center text-center "
    data-testid="error-container"
  >
    <span
      class="text-red-500 mr-3"
      data-testid="error-message"
    >
      Test error message
    </span>
    <button
      aria-label="Dismiss error"
      class="
                    text-3xl text-red-300 hover:text-[var(--color-primary)]
                    transition-colors duration-300 cursor-pointer
                "
      data-testid="error-dismiss-button"
    >
      Ã—
    </button>
  </div>
</DocumentFragment>
`;
